aws/sdk/examples/dynamodb/src/bin/crud.rs

```diff
@@ -142,91 +141,21 @@ async fn query(client: &dynamodb::Client, item: Item) {
-    let mut found_match = true;
-    let mut cond = HashMap::new();
-    cond.insert("#key".to_string(), key.to_string());
-    let mut expr = HashMap::new();
-    expr.insert(":value".to_string(), user_av);
-    match client
+
+    let resp = client
         .query()
         .table_name(item.table)
-        .set_key_condition_expression(Some("#key = :value".to_string()))
-        .set_expression_attribute_names(Some(cond))
-        .set_expression_attribute_values(Some(expr))
+        .key_condition_expression("#key = :value".to_string())
+        .expression_attribute_names("#key".to_string(), key.to_string())
+        .expression_attribute_values(":value".to_string(), user_av)
         .select(Select::AllAttributes)
         .send()
-        .await
-    {
-        Ok(resp) => {
-            let items = resp.items.unwrap_or_default();
-            for item in items {
-                // Do key values match?
-                match item.get(&String::from(key)) {
-                    None => found_match = false,
-                    Some(v) => {
-                        if v != &userav {
-                            found_match = false;
-                        }
-                    }
-                }
-
-                // Do age values match?
-                match item.get(&String::from("age")) {
-                    None => found_match = false,
-                    Some(v) => {
-                        if v != &age_av {
-                            found_match = false;
-                        }
-                    }
-                }
-
-                // Do first name values match?
-                match item.get(&String::from("first_name")) {
-                    None => found_match = false,
-                    Some(v) => {
-                        if v != &first_av {
-                            found_match = false;
-                        }
-                    }
-                }
-
-                // Do last name values match?
-                match item.get(&String::from("last_name")) {
-                    None => found_match = false,
-                    Some(v) => {
-                        if v != &last_av {
-                            found_match = false;
-                        }
-                    }
-                }
-
-                // Do account type values match?
-                match item.get(&String::from("account_type")) {
-                    None => found_match = false,
-                    Some(v) => {
-                        if v != &type_av {
-                            found_match = false;
-                        }
-                    }
-                }
-            }
-        }
-        Err(e) => {
-            println!("Got an error querying the table:");
-            println!("{}", e);
-            process::exit(1);
-        }
-    };
+        .await;
 
-    if !found_match {
-        println!("Did not find matching entry in table");
+    if resp.unwrap().count > 0 {
```

Comment:
this unwrap should be `?`

Reply:
I'll change it.

Reply:
When I try:

if resp?.count > 0

I get:
````
the `?` operator can only be used in an async function that returns `Result` or `Option` (or another type that implements `std::ops::Try`)

cannot use the `?` operator in an async function that returns `()`

help: the trait `std::ops::Try` is not implemented for `()`
note: required by `std::ops::Try::from_error`
```

So I reverted back to unwrap.